### PR TITLE
Feat/support v8

### PR DESCRIPTION
**Pull request by jelugbo**
Update to work with WDIO v8

### CODE REVIEW

**Comment by dlitmano** [comment]
Hey, can you run all checks inside pipeline.

**Comment by jelugbo** [comment]
Workflow is triggered on PR of Push to main branch. Did npm run test on local and all test passed

**Comment by bjelugbo** [comment]
@christian-bromann, comments have been addressed in new commit

File: babel.config.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ module.exports = {
         [ '@babel/preset-env',
             {
                 targets: {
-                    node: 8
+                    node: 16
```
Comment (christian-bromann) [inline babel.config.js:6]:
```suggestion
                    node: 18
```

File: lib/reporter.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
+// import * as fso from 'fs';
+
+import { readFileSync , existsSync} from 'fs'
+//import { outputFile, outputFileSync } from 'fs-extra/esm'
```
Comment (christian-bromann) [inline lib/reporter.ts:4]: ?

File: package.json [hunk 3]
```diff
@@ -43,40 +43,41 @@
...
         "cucumber-messages": "^8.0.0",
         "eslint": "^8.30.0",
         "eslint-config-prettier": "^8.5.0",
         "eslint-plugin-import": "^2.26.0",
         "husky": "^8.0.2",
-        "jest": "^29.3.1",
         "prettier": "2.8.1",
         "release-it": "^15.5.1",
         "rimraf": "^3.0.2",
         "ts-jest": "^29.0.3",
```
Comment (christian-bromann) [inline package.json:76]: no need for this anymore

File: package.json [hunk 4]
```diff
@@ -43,40 +43,41 @@
...
+        "webdriverio": "~8.6.9"
     },
     "peerDependencies": {
         "@wdio/cucumber-framework": "8.x"
     },
     "devDependencies": {
         "@tsconfig/node18": "^1.0.1",
         "@types/fs-extra": "^9.0.13",
-        "@types/jest": "^29.2.4",
+        "@types/jest": "^29.5.0",
```
Comment (christian-bromann) [inline package.json:60]: same

File: vitest.config.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/// <reference types="vitest" />
```
Comment (christian-bromann) [inline vitest.config.ts:1]: Let's remove `jest.config.js`

File: lib/tests/fileExists.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
-import { accessSync } from 'fs-extra';
+import { accessSync } from 'fs';
 
 /**
  * Check if a file exists
  */
 export const fileExists = ( filePath: string ): boolean => {
+   // const fse = require('fs-extra');
```
Comment (christian-bromann) [inline lib/tests/fileExists.ts:7]:
```suggestion
```

File: vitest.config.ts (hunk 5 above)
Comment (bjelugbo) [reply vitest.config.ts:1]: jest.config.js is removed

File: package.json (hunk 3 above)
Comment (bjelugbo) [reply package.json]: removed

File: package.json (hunk 4 above)
Comment (bjelugbo) [reply package.json]: removed as well

File: lib/metadata.ts [hunk 7]
```diff
@@ -17,12 +16,18 @@ export class Metadata {
...
+        const w3cCaps: cjson_metadata | undefined
+            = Object.prototype.hasOwnProperty.call(data.config.capabilities, 'cjson:metadata') ? // Fixes: https://github.com/webdriverio-community/wdio-cucumberjs-json-reporter/issues/73 Fallback
+            (data.config.capabilities as cjson_metadata)
             : (browser.options as WebdriverIOExtended)?.requestedCapabilities?.cjson_metadata;
+
+      /*   const w3cCaps: cjson_metadata | undefined =
+            !Object.prototype.hasOwnProperty.call( data.config.capabilities, 'cjson:metadata' ) ?
+                ( browser.options as WebdriverIOExtended )?.requestedCapabilities?.cjson_metadata :
+                data.config.capabilities['cjson:metadata'] as cjson_metadata; */
+
```
Comment (christian-bromann) [inline lib/metadata.ts:30]:
```suggestion
```

File: lib/reporter.ts [hunk 8]
```diff
@@ -1,6 +1,9 @@
+// import * as fso from 'fs';
+
```
Comment (christian-bromann) [inline lib/reporter.ts:2]:
```suggestion
```

File: lib/reporter.ts [hunk 9]
```diff
@@ -317,7 +320,8 @@ class CucumberJsJsonReporter extends WDIOReporter {
      * Get the current scenario
      */
     public getCurrentScenario(): Scenario {
-        return this.report.feature.elements[this.report.feature.elements.length - 1];
+        console.log('=>',JSON.stringify(this.report))
```
Comment (christian-bromann) [inline lib/reporter.ts:323]:
```suggestion
```

File: lib/reporter.ts [hunk 10]
```diff
@@ -326,7 +330,7 @@ class CucumberJsJsonReporter extends WDIOReporter {
     public getCurrentStep(): Step {
         const currentScenario = this.getCurrentScenario();
 
-        return currentScenario.steps[currentScenario.steps.length - 1];
+        return currentScenario.steps![currentScenario.steps!.length - 1];
```
Comment (christian-bromann) [inline lib/reporter.ts:333]: What happens if `.step` is undefined?

File: lib/reporter.ts [hunk 11]
```diff
@@ -26,8 +27,8 @@ import {
 import { HookStatsExtended, SuiteStatsExtended, TestStatsExtended } from './types/wdio.js';
 import WDIOReporter, { HookStats, RunnerStats, SuiteStats, TestStats } from '@wdio/reporter';
 import { CucumberJsJsonReporterInterface } from './types.js';
-import { Metadata } from './metadata.js';
-import Utils from './utils.js';
+import { Metadata } from './metadata';
+import Utils from './utils';
```
Comment (christian-bromann) [inline lib/reporter.ts:31]: ESM packages require file extensions for imports.

```suggestion
import { Metadata } from './metadata.js';
import Utils from './utils.js';
```

File: lib/reporter.ts (hunk 10 above)
Comment (bjelugbo) [reply lib/reporter.ts]: Code updated to check if step is undefined

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by christian-bromann** [comment]
Thanks for the work @jelugbo , let's continue convo in  #113

**Pull request merged**